toolchain/lower/file_context.cpp

```diff
@@ -308,20 +332,37 @@ auto FileContext::BuildFunctionDefinition(SemIR::FunctionId function_id)
     return;
   }
 
-  llvm::Function* llvm_function = GetFunction(function_id);
-  if (!llvm_function) {
-    // We chose not to lower this function at all, for example because it's a
-    // generic function.
-    return;
+  llvm::Function* llvm_function;
+  if (!specific_id.has_value()) {
+    llvm_function = GetFunction(function_id);
+    if (!llvm_function) {
+      // We chose not to lower this function at all, for example because it's a
+      // generic function.
+      return;
+    }
+  } else {
+    llvm_function = specific_functions_[specific_id.index];
   }
 
-  FunctionContext function_lowering(*this, llvm_function,
+  // For non-generics we do not lower. For generics, the llvm function was
+  // created via GetOrCreateFunction prior to this when building the
+  // declaration.
+  CARBON_CHECK(llvm_function, "The llvm function must exist");
+
+  BuildFunctionBody(function_id, function, llvm_function, specific_id);
+}
+
+auto FileContext::BuildFunctionBody(SemIR::FunctionId function_id,
+                                    const SemIR::Function& function,
+                                    llvm::Function* llvm_function,
+                                    SemIR::SpecificId specific_id) -> void {
+  const auto& body_block_ids = function.body_block_ids;
+  CARBON_DCHECK(llvm_function && !body_block_ids.empty());
```

Comment:
```suggestion
  CARBON_DCHECK(llvm_function);
  CARBON_DCHECK(!body_block_ids.empty());
```
So you know which one failed

Reply:
Oh I see this came from https://github.com/carbon-language/carbon-lang/pull/5014, moving there.